vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3174,6 +3199,35 @@ private void generateStyleData(T item, JsonObject jsonObject) {
         }
     }
 
+    private void generateRowsDragAndDropAccess(T item, JsonObject jsonObject) {
+        if (this.getDropMode() != null) {
+            Boolean dropEnabled = this.getDropMode() != null
+                    && dropFilter.apply(item);
+            if (!dropEnabled) {
```

Comment:
There's duplicate `getDropMode() != null` above. Also, no need for two nested `if`-blocks.

This could be simplified to e.g.:
```java
if (getDropMode() != null && !dropFilter.apply(item)) {
  jsonObject.put
```

Reply:
Done.